File: APP03022-mrch.backend.somx.catalogos-api/src/main/java/com/sodimac/catman/api/mapper/CatalogDetailMapper.java
package com.sodimac.catman.api.mapper;

import java.util.List;
import java.util.Map;

import org.springframework.stereotype.Component;

import com.sodimac.catman.api.model.dto.CatalogDetailDto;
import com.sodimac.catman.api.model.entity.CatalogDetail;

@Component
public class CatalogDetailMapper {

    public CatalogDetailDto toDto(CatalogDetail entity, Map<Integer, String> dictionary) {
        if (entity == null) {
            return null;
        }

        return CatalogDetailDto.builder()
                .key(entity.getKey())
                .internalStatus(entity.getInternalStatus())
                .externalKey(entity.getExternalKey())
                .value(entity.getValue())
                .description(dictionary.getOrDefault(entity.getDictId(), ""))
                .color(entity.getColor())
                .sortOrder(entity.getSortOrder())
                .validFrom(entity.getValidFrom())
                .validTo(entity.getValidTo())
                .attributes(entity.getAttributes())
                .build();
    }

    public CatalogDetail toEntity(CatalogDetailDto dto, Integer dictId) {
        if (dto == null) {
            return null;
        }

        return CatalogDetail.builder()
                .key(dto.getKey())
                .externalKey(dto.getExternalKey())
                .value(dto.getValue())
                .dictId(dictId)
                .color(dto.getColor())
                .sortOrder(dto.getSortOrder())
                .validFrom(dto.getValidFrom())
                .validTo(dto.getValidTo())
                .attributes(dto.getAttributes())
                .build();
    }

    public List<CatalogDetailDto> toDtoList(List<CatalogDetail> entities, Map<Integer, String> dictionary) {
        if (entities == null) {
            return null;
        }

        return entities.stream()
                .map(e -> toDto(e, dictionary))
                .toList();
    }
}

File: APP03022-mrch.backend.somx.util-api/src/mappers/catalogDetail.mapper.ts
import type { CatalogDetail } from '@/entities/CatalogDetail.entity.js';
import type { CatalogDetailDto } from '@/dto/catalog.dto.js';

export function toDto(
    entity: CatalogDetail | null | undefined,
    dictionary: Map<number, string>
): CatalogDetailDto | null {
    if (!entity) return null;
    const description = dictionary.get(entity.dictId) ?? '';
    return {
        key: entity.key,
        internalStatus: entity.internalStatus ?? null,
        externalKey: entity.externalKey ?? null,
        value: entity.value ?? null,
        description,
        color: entity.color ?? null,
        sortOrder: entity.sortOrder ?? null,
        validFrom: entity.validFrom ?? null,
        validTo: entity.validTo ?? null,
        attributes: entity.attributes ?? null
    };
}

export function toDtoList(
    entities: CatalogDetail[] | null | undefined,
    dictionary: Map<number, string>
): CatalogDetailDto[] {
    if (!entities) return [];
    return entities
        .map(e => toDto(e, dictionary))
        .filter((d): d is CatalogDetailDto => d !== null);
}

